sensor/kubernetes/eventpipeline/pipeline.go

```diff
@@ -7,25 +7,34 @@ import (
 	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/sensor/common"
 	"github.com/stackrox/rox/sensor/common/config"
 	"github.com/stackrox/rox/sensor/common/detector"
 	"github.com/stackrox/rox/sensor/kubernetes/client"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
 	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/output"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/resolver"
 	"github.com/stackrox/rox/sensor/kubernetes/listener"
 )
 
 // New instantiates the eventPipeline component
 func New(client client.Interface, configHandler config.Handler, detector detector.Detector, nodeName string, resyncPeriod time.Duration, traceWriter io.Writer) common.SensorComponent {
 	// TODO(ROX-13413): Move this env.EventPipelineOutputQueueSize to CreateOptions
 	outputQueue := output.New(detector, env.EventPipelineOutputQueueSize.IntegerSetting())
-	resourceListener := listener.New(client, configHandler, nodeName, resyncPeriod, traceWriter, outputQueue)
+	var depResolver component.Resolver
+	depResolver = outputQueue
+	if features.ResyncDisabled.Enabled() {
+		depResolver = resolver.New(outputQueue)
+	}
+	resourceListener := listener.New(client, configHandler, nodeName, resyncPeriod, traceWriter, depResolver)
```

Comment:
nit: I would prefer to be explicit about how we are instantiating the resource listeners instead of trying to save a few lines of code:

```golang
var resourceListener component.PipelineComponent
if features.ResyncDisabled.Enabled() {
        depResolver = resolver.New(outputQueue)
	resourceListener = listener.New(client, configHandler, nodeName, resyncPeriod, traceWriter, depResolver)
} else {
        resourceListener = listener.New(client, configHandler, nodeName, resyncPeriod, traceWriter, outputQueue)
}
```

I think it's more clear to someone that isn't one of us to know which dependencies are passed based on the feature flag, at the cost of repeating a line.

Reply:
Agree. I've changed it accordingly